Clarify migration process to gatsby-link 📝

whmountains opened the pull request:
A [discussion](https://discordapp.com/channels/102860784329052160/103314369600843776?jump=346651787824201730) came up on discord about the how to get the functionality of `<IndexLink>` with `gatsby-link`.

I was all fired up to modify the gatsby-link package and submit a PR when I realized that `<IndexLink>` was dropped in react-router v4 and the new way to do it is by passing the `exact` prop. 🤦‍♀️   So instead of modifying `gatsby-link` I'm (attempting to) improve the documentation. 😄

KyleAMathews commented on docs/docs/migrating-from-v0-to-v1.md:
```diff
@@ -24,24 +24,24 @@ git mv utils src
+<Link to="/page-2/" activeClassName="selected">Page 2</Link>
 
+// `exact` prop replaces <IndexLink> from react-router v3
+<Link to="/" exact>Home</IndexLink>
```
IndexLink slipped in

KyleAMathews commented on docs/docs/migrating-from-v0-to-v1.md:
```diff
@@ -24,24 +24,24 @@ git mv utils src
+<Link to="/page-2/">Page 2</Link>
 
-Use `gatsby-link` everywhere and things will Just Work™.
+// Equivelant to react-router's <NavLink>
```
Spell check

KyleAMathews commented on docs/docs/migrating-from-v0-to-v1.md:
```diff
@@ -24,24 +24,24 @@ git mv utils src
-
-Prefixing links is also now handled automatically by our new `<Link>` component
-so remove usages of `prefixLink` in links.
+// Equivelant to react-router's <Link>
```
Here too

KyleAMathews commented on docs/docs/migrating-from-v0-to-v1.md:
```diff
@@ -24,24 +24,24 @@ git mv utils src
 ## Replace react-router's Link component with gatsby-link
 
-Add new package to project.
+`gatsby-link` is a wrapper for the `<Link>` component in react-router.  It automatically prefixes urls and handles prefetching on browsers that don't support ServiceWorker.  Gatsby doesn't include it, so you should install it:
```
A nit, it prefetches regardless of whether you're using a sw or not.

Also change the last sentence to something like:  "Add `gatsby-link` to your project by running:"

whmountains commented:
Ooof, I can't believe I let so many typos slip in!

whmountains commented:
@KyleAMathews How's that?

* Dropped the mention of ServiceWorker.
* Changed last sentence of first paragraph to "Add `gatsby-link` to your project by running:"
* Equivelant -> Equivalent
* Fixed mismatched closing JSX tag.

KyleAMathews commented on docs/docs/migrating-from-v0-to-v1.md:
```diff
@@ -24,24 +24,24 @@ git mv utils src
 ## Replace react-router's Link component with gatsby-link
 
-Add new package to project.
+`gatsby-link` is a wrapper for the `<Link>` component in react-router.  It automatically prefixes urls and handles prefetching.  Add `gatsby-link` to your project by running:
```
Because automatically prefixing is quite different than in 0.x — make that more empasized. Perhaps keep this line but also restore the "Prefixing links is also now handled automatically by our new `<Link>` component so remove usages of `prefixLink` in links."

whmountains replied:
Yes, I debated about that a lot myself.  I will add it back.

KyleAMathews commented:
Thanks for the docs cleanups!
